test/algorithms/classifiers/test_neural_network_classifier.py

```diff
@@ -72,17 +76,22 @@ def test_classifier_with_opflow_qnn(self, config):
 
         if opt == "bfgs":
             optimizer = L_BFGS_B(maxiter=5)
-        else:
+        elif opt == "cobyla":
             optimizer = COBYLA(maxiter=25)
+        else:
+            optimizer = None
 
         num_inputs = 2
         ansatz = RealAmplitudes(num_inputs, reps=1)
         qnn = TwoLayerQNN(num_inputs, ansatz=ansatz, quantum_instance=quantum_instance)
         initial_point = np.array([0.5] * ansatz.num_parameters)
 
-        classifier = NeuralNetworkClassifier(
-            qnn, optimizer=optimizer, loss=loss, initial_point=initial_point
-        )
+        if optimizer is None:
+            classifier = NeuralNetworkClassifier(qnn, loss=loss, initial_point=initial_point)
+        else:
+            classifier = NeuralNetworkClassifier(
+                qnn, optimizer=optimizer, loss=loss, initial_point=initial_point
+            )
```

Comment:
I guess there's no need to have this if statement. Just passing all parameters including `optimizer` which is set to `None` is enough.

Reply:
Thought to make the need of these test explicit, but I definitely agree it does not make sense to keep them.